### PR TITLE
Work on issue #154

### PR DESCRIPTION
**Pull request by anikitenko**
Started my work on issue #154. Currently only "mv {boot,mkfs,staging3} to #153" is done. I'll continue working.. ready for comments..)

### CODE REVIEW

File: lepton/const.go [hunk 1]
```diff
@@ -21,19 +21,22 @@ const manifest string = `(
 const Version = "0.2"
 const OpsReleaseUrl = "https://storage.googleapis.com/cli/%v/ops"
 
+// .ops directory
+const OpsDir string = ".ops"
```
Comment (tijoytom) [inline lepton/const.go]: This should be use $HOME/.ops

File: lepton/image.go [hunk 2]
```diff
@@ -249,62 +256,82 @@ func (bc dummy) Write(p []byte) (int, error) {
 	return len(p), nil
 }
 
-func DownloadBootImages() error {
```
Comment (tijoytom) [inline lepton/image.go]: Dont delete this function, it's used else where.

File: lepton/image.go [hunk 3]
```diff
@@ -249,62 +256,82 @@ func (bc dummy) Write(p []byte) (int, error) {
...
 
 	fmt.Println("Downloading..", filepath)
 	out, err := os.Create(filepath + ".tmp")
 	if err != nil {
 		return err
 	}
-	defer out.Close()
-	
+	defer func() {
+		if err := out.Close(); err != nil {
```
Comment (tijoytom) [inline lepton/image.go]: Probably no need to panic here, Close returns any errors from the underlying listener. I am fine with just out.Close and not checking for error here.

File: lepton/const.go (hunk 1 above)
Comment (anikitenko) [reply lepton/const.go]: Extended a little use of constant OpsDir. It's needed to determine "base" directory in HOME of user. For example, path to `packages` directory is `path.Join(home, OpsDir, "packages")`

File: lepton/image.go (hunk 2 above)
Comment (anikitenko) [reply lepton/image.go]: I didn't delete this function. I just extended it to make sure that `OpsDir` exists and it's name is the same `DownloadBootImages`. I executed tests and checked for use of that function - everything is alright

File: lepton/image.go (hunk 3 above)
Comment (anikitenko) [reply lepton/image.go]: Agree. I removed panics in defer functions but left as following to silence editor warnings like "Unhandled error..":
`defer func() {
		_ = out.Close()
	}()`
and
`defer func() {
		_ = resp.Body.Close()
	}()`

Comment (eyberg) [reply lepton/image.go]: yeh i mean those warnings are there for a reason - i'd prefer we handle all errors correctly - it's a go idiom (and just good programming as well)

Comment (anikitenko) [reply lepton/image.go]: and you're right too.. there are no panics in that func but are returns.. so should I update that to `return` if `Close` returns error?

Comment (tijoytom) [reply lepton/image.go]: Editors are just dump, they don't have any contextual information. So I won't worry too much about some editor plugin warning about it. But anyways just return the error and yes never panic inside the exported function. Let the caller handle it.

Comment (anikitenko) [reply lepton/image.go]: Got it, updating PR..)

File: lepton/image.go (hunk 2 above)
Comment (tijoytom) [reply lepton/image.go]: You renamed it right? What I was telling is that it's used outside of this repository. So you won't see any test failure, remember this is a library shared across projects

Comment (anikitenko) [reply lepton/image.go]: Got it, returned back function with it's parameters

File: lepton/image.go (hunk 3 above)
Comment (anikitenko) [reply lepton/image.go]: I updated PR with reworked defer funcs

**Pull request closed without merging**